library/core/src/marker.rs

```diff
@@ -1006,3 +1006,17 @@ pub trait FnPtr: Copy + Clone {
     #[lang = "fn_ptr_addr"]
     fn addr(self) -> *const ();
 }
+
+/// An experimental auto trait with default bounds
+#[fundamental]
+#[rustc_unsafe_specialization_marker]
+#[unstable(feature = "default_auto_traits", issue = "none")]
+#[cfg_attr(not(bootstrap), lang = "trait1")]
+pub unsafe auto trait Leak {}
+
+/// An experimental auto trait with default bounds
+#[fundamental]
+#[rustc_unsafe_specialization_marker]
+#[unstable(feature = "default_auto_traits", issue = "none")]
+#[cfg_attr(not(bootstrap), lang = "trait2")]
+pub unsafe auto trait SyncDrop {}
```

Comment:
we already have https://doc.rust-lang.org/std/marker/trait.Destruct.html which fulfills the same purpose

Reply:
If `Destruct` is always going to be implemented by compiler, with no manual implementations (including negative manual implementation), then it may fine to reuse something that is not an auto-trait, like `Destruct`.

I'm not sure how it will work in (coinductive?) cycles in the resolver though.
Some minor issues like `dyn Trait + Destruct` not working due to `Destruct` not being an auto trait are also possible.

These are all the questions that I wanted to avoid in this PR though, by not adding anything to libcore at this step (https://github.com/rust-lang/rust/pull/120706#discussion_r1521470905).